demisto_sdk/commands/common/hook_validations/id.py

```diff
@@ -94,6 +92,197 @@ def _is_incident_type_default_playbook_found(self, incident_type_data):
 
         return is_valid
 
+    def _is_incident_field_scripts_found(self, incident_field_data, incident_field_file_path=None):
+        """Check if scripts and field calculations scripts of an incident field is in the id_set
+
+        Args:
+            incident_field_data (dict): Dictionary that holds the extracted details from the given incident field.
+            incident_field_file_path (str): Path to the file.
+
+        Returns:
+            bool. Whether the scripts are in the id_set or not.
+        """
+        is_valid = True
+        scripts_not_in_id_set = set()
+
+        incident_field_id = list(incident_field_data.keys())[0]
+        incident_field = incident_field_data.get(incident_field_id, {})
+        incident_field_name = incident_field.get('name', incident_field_id)
+        scripts_set = set(incident_field.get('scripts', []))
+
+        # Check if the incident field scripts are in the id_set:
+        if scripts_set:
+            scripts_not_in_id_set = self._get_scripts_that_are_not_in_id_set(scripts_set)
+
+        # Add error message if there are scripts that aren't in the id_set:
+        if scripts_not_in_id_set:
+            is_valid = False
+            scripts_not_in_id_set_str = ', '.join(scripts_not_in_id_set)
+            error_message, error_code = Errors.incident_field_non_existent_script_id(incident_field_name,
+                                                                                     scripts_not_in_id_set_str)
+            if not self.handle_error(error_message, error_code, file_path=incident_field_file_path):
+                is_valid = True
+
+        return is_valid
+
+    def _is_layouts_container_scripts_found(self, layouts_container_data, layouts_container_file_path=None):
+        """Check if scripts of a layouts container is in the id_set
+
+        Args:
+            layouts_container_data (dict): Dictionary that holds the extracted details from the given layouts container.
+            layouts_container_file_path (str): Path to the file.
+
+        Returns:
+            bool. Whether the scripts are in the id_set or not.
+        """
+        is_valid = True
+        scripts_not_in_id_set = set()
+
+        layouts_container_id = list(layouts_container_data.keys())[0]
+        layouts_container = layouts_container_data.get(layouts_container_id, {})
+        layouts_container_name = layouts_container.get('name', layouts_container_id)
+        layouts_container_tabs = self._get_layouts_container_tabs(layouts_container)
+        scripts_set = set(get_layouts_scripts_ids(layouts_container_tabs))
```

Comment:
we should really consider adding a `scripts` key for the info kept about layout containers - like we have for layouts.
@ShacharKidor - please open an issue about this and point to this code as to how the extracted set should be done.

@tomneeman151293 - FYI

Reply:
open issue: 
[https://github.com/demisto/etc/issues/41277](url)